src/auth/index.js

```diff
@@ -11,27 +11,57 @@ const jwtDecode = require('jwt-decode');
 
 const EMBEDDED_USER_AUTH_DATA_PARTS = 4;
 
-export default class Auth {
+export class AuthFactory {
+  constructor() {
+    throw new StitchError('StitchClient can only be made from the StitchClientFactory.create function');
+  }
+
+  static create(client, rootUrl, options) {
+    return new Promise((resolve, reject) => {
+      resolve(new Auth(client, rootUrl, options)._willInitialize);
+    });
+  }
+}
+
+export class Auth {
   constructor(client, rootUrl, options) {
-    let namespace;
-    if (!client || client.clientAppID === '') {
-      namespace = 'admin';
-    } else {
-      namespace = `client.${client.clientAppID}`;
-    }
+    this._willInitialize = new Promise((resolve, reject) => {
```

Comment:
[minor] I think we discussed this (at couches). I'd prefer if there were no way to construct any of these "concrete" classes. The `_willInitialize` concept across the two clients and auth feels kind of awkward that we do a `new` + `. _willInitialize `. What do you think of doing a separate function that is not exported that does `Object.create(Class.prototype)` and then taking the body of this here and appending to that?

Then you could write in the `create` functions: `resolve(createAuth())`